git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,810 @@
+  2. The second line tells exactly which versions of the file Git is comparing; `df0654a` and `315bf3a` are unique computer-generated labels for those versions.
+  3. The third and fourth lines once again show the name of the file being changed.
+  4. The remaining lines are the most interesting, they show us the actual differences and the lines on which they occur. In particular, the `+` marker in the first column shows where we added a line.
+</div>
+
+<div class = 'warning'>
+If we immediately commit this change using `git commit` what will happen?
+
+Git won't commit because we didn't use `git add` first! Git will give you a message reminding you to use `git add`.
+
+</div>
+
+
+Let's try adding the file and then committing the changes:
+
+```
+$ git add mars.txt
+$ git commit -m "Add concerns about effects of Mars' moons on Wolfman"
+
+[main 34961b1] Add concerns about effects of Mars' moons on Wolfman
+ 1 file changed, 1 insertion(+)
+```
+
+Double check that everything worked by running `git status`:
+
+```
+$ git status
+
+On branch main
+
+nothing to commit, working directory clean
+```
+
+### How often should you `commit`?
+
+Just like any method of backing up or saving your work, the benefits of using Git for version control are only available if you regularly use them. In our example, we have been committing every time we add or change a line in a single file. That is going to be odious overkill for most projects.
+
+- **Commit each chunk of changes** If you add a new section to a file, or change a certain item across multiple files, it is a good practice to commit those changes with a message describing them. If a commit message like `-m "add section on atmosphere"` or `-m "reorder paragraphs on moons from smallest to largest moon"` accurately describes your changes, make sure you commit the changes so that you can find them again later!
+
+- **Commit before making any changes that could break your code** If your code currently works are you intend to add something new that may break it, make sure you commit so that you will be able to revert to the working version if needed.
+
+- **Commit regularly** Even if you are only making very small changes, don't forget to commit at regular intervals, like before you switch tasks or end your work day.
+
+### Quiz: `add` and `commit`
+
+1. What sequence of commands would track the changes you made to `myFile.txt`?
+
+  a. `$ git add myFile.txt`
+
+  b. `$ git commit myFile.txt`
+
+  c. `$ git add -m "my short, descriptive message"`
+
+  d. `$ git commit -m "my short, descriptive message"`
+
+[( )] First run a, then b
+[(X)] First run a, then d
+[( )] First run c, then b
+[( )] First run c, then d
+***
+<div class = 'answer'>
+You must run the `git add` to tell Git to stage `myFile.txt`, then you can commit the changes and leave your short, descriptive message. We will discuss how to think of these steps in the [next section](#understanding-the-git-workflow).
```

Comment:
This isn't really the next section anymore

Reply:
fixed!